diff --git a/modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt b/modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt
index ca5092c0c6..0bcd8816c4 100644
--- a/modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt
+++ b/modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt
@@ -32,7 +32,11 @@ zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1
 zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1c_dir}/include/ip)
 zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1c_dir}/source/cy_device.c)
 
-zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_ext.S)
+if(CONFIG_ARMCLANG)
+  zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_A_Clang/cy_syslib_ext.S)
+else()
+  zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_ext.S)
+endif()
 
 # Peripheral drivers
 zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6_LEGACY      ${pdl_drv_dir}/source/cy_sysint.c)
